src/libcrun/criu.c

```diff
@@ -33,22 +33,53 @@
 #  include "linux.h"
 #  include "status.h"
 #  include "utils.h"
+#  include "cgroup.h"
 
 #  define CRIU_CHECKPOINT_LOG_FILE "dump.log"
 #  define CRIU_RESTORE_LOG_FILE "restore.log"
 #  define DESCRIPTORS_FILENAME "descriptors.json"
 
+static const char *console_socket = NULL;
+
+static int
+criu_notify (char *action, criu_notify_arg_t na)
+{
+  if (strncmp (action, "orphan-pts-master", 17) == 0)
```

Comment:
@adrianreber do you think that inverting the logic of this check would make the function look better?  For example:

```
if (strncmp (action, "orphan-pts-master", 17) != 0)
  return 0;

// function flow proceeds...
```

This way, the entire logic doesn't need to be nested.  And it's similar to what is done within the `if` when checking `console_socket`, `console_socket_fd`, and `ret`.

Reply:
Usually I try to structure the code like you suggested. As the notify function potentially can handle different callback types it does not sound useful once there are multiple callbacks. Right now, however, I can changed the code flow. You are right. I will do that as I want to include additional tests for this PR anyway.